File: sdks/python/examples/generate_image.py
"""Example: generate an image with the LiteGen Python SDK."""
import os

from litegen import LiteGenClient


def main() -> None:
    client = LiteGenClient(
        base_url=os.environ.get("LITEGEN_BASE_URL", "http://localhost:4000"),
        api_key=os.environ.get("LITEGEN_API_KEY"),
    )
    result = client.images.generate(
        prompt="a serene mountain landscape at sunset, oil painting",
        model="openai/dall-e-3",
        size="1024x1024",
        quality="hd",
    )
    print("Image URL:", result["data"][0].get("url"))
    print("Provider:", result["provider"])
    if result.get("usage"):
        print("Cost USD:", result["usage"]["cost_usd"])


if __name__ == "__main__":
    main()

File: sdks/typescript/examples/generate-image.ts
import { LiteGenClient } from "../src";

const client = new LiteGenClient({
  baseUrl: process.env.LITEGEN_BASE_URL ?? "http://localhost:4000",
  apiKey: process.env.LITEGEN_API_KEY,
});

const result = await client.images.generate({
  prompt: "a serene mountain landscape at sunset, oil painting",
  model: "openai/dall-e-3",
  size: "1024x1024",
  quality: "hd",
});

console.log("Image URL:", result.data[0]?.url);
console.log("Provider:", result.provider);
if (result.usage) {
  console.log("Cost USD:", result.usage.cost_usd);
}
